src/ray/object_manager/object_manager.cc

```diff
@@ -586,7 +586,15 @@ void ObjectManager::SubscribeRemainingWaitObjects(const UniqueID &wait_id) {
 }
 
 void ObjectManager::WaitComplete(const UniqueID &wait_id) {
-  auto &wait_state = active_wait_requests_.find(wait_id)->second;
+  auto iter = active_wait_requests_.find(wait_id);
+  if (iter == active_wait_requests_.end()) {
```

Comment:
Is the issue still present when this check is moved within the timer (right above line 582)? I think the issue will be easier to identify if we have the check there. If we can do that, and the issue is no longer present, then I will approve and merge!